pipeline.py

```diff
@@ -72,31 +90,96 @@ class TextProcessingPipeline:
     """
 
     def __init__(self, corpus_manager: CorpusManager):
-        pass
+        self.corpus_manager = corpus_manager
 
     def run(self):
         """
         Runs pipeline process scenario
         """
-        pass
+
+        for article in self.corpus_manager.get_articles().values():
+            article_raw_text = article.get_raw_text()
+
+            cleaned_tokens = []
+            single_tagged_tokens = []
+            multiple_tagged_tokens = []
+
+            for token in self._process(article_raw_text):
+                cleaned_tokens.append(token.get_cleaned())
+                single_tagged_tokens.append(token.get_single_tagged())
+                multiple_tagged_tokens.append(token.get_multiple_tagged())
+
+            article.save_as(' '.join(cleaned_tokens), ArtifactType.cleaned)
+            article.save_as(' '.join(single_tagged_tokens), ArtifactType.single_tagged)
+            article.save_as(' '.join(multiple_tagged_tokens), ArtifactType.multiple_tagged)
 
     def _process(self, raw_text: str):
         """
         Processes each token and creates MorphToken class instance
         """
-        pass
+        pattern = re.compile(r'[а-яА-Яa-zA-z ё]')
+        for symbol in raw_text:
+            if not pattern.match(symbol):
+                text = raw_text.replace(symbol, '')
+
+        analyzed_text = Mystem().analyze(text)
+        morph = pymorphy2.MorphAnalyzer()
+        tokens = []
+        for token in analyzed_text:
+            if 'analysis' not in token:
+                continue
+            if not token['analysis']:
+                continue
+
+            morph_token = MorphologicalToken(original_word=token['text'])
+            morph_token.normalized_form = token['analysis'][0]['lex']
+            morph_token.tags_mystem = token['analysis'][0]['gr']
+            morph_token.tags_pymorphy = morph.parse(token['text'])[0].tag
+            tokens.append(morph_token)
+        return tokens
 
 
 def validate_dataset(path_to_validate):
     """
     Validates folder with assets
     """
-    pass
+    path = Path(path_to_validate)
+
+    if not path.exists():
+        raise FileNotFoundError
+
+    if not path.is_dir():
+        raise NotADirectoryError
+
+    if not list(path.iterdir()):
+        raise EmptyDirectoryError
+
+    number_raw_txt = 0
+    number_meta = 0
+
+    for file in sorted(path.glob('*'), key=lambda x: int(x.name[:x.name.find('_')])):
+
+        if file.name.endswith('raw.txt'):
+            number_raw_txt += 1
+            if f'{number_raw_txt}_raw' not in file.name:
+                raise InconsistentDatasetError
+
+            with open(file, 'r', encoding='utf-8') as the_file:
```

Comment:
where did you find a requirement to open the file for emptiness? If this is not stated in docs, remove. Otherwise, please point to that place. but if you insist, there is a much smarter check that file is empty. How do you think what am I talking about?

Reply:
i removed it. 
the file size should be checked. if the size = 0, the file is empty

Reply:
done